File: Week_01/OnePreDay/AddOne-66.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
    let i = digits.length - 1;
    let tempCarryNum = 1;
    while (i >= 0) {
        digits[i] = digits[i] + tempCarryNum;
        if (digits[i] >= 10) {
            digits[i] = digits[i] % 10;
            tempCarryNum = 1;
        } else {
            tempCarryNum = 0
        }
        i--;
    }
    if (tempCarryNum == 1) {
        digits.unshift(1);
    }
    return digits;
};
console.log(plusOne([1,1,7,9]));

File: Week_01/OnePreDay/AddOne-66.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        i = len(digits)
        addNum = 1
        while (i >= 0):
            digits[i] = digits[i] + addNum
            if (digits[i] >= 10):
                digits[i] = digits[i] % 10;
                tempCarryNum = 1
            else:
                tempCarryNum = 0
            i -= 1
        if addNum == 1:
            digits = [1] + digits
        return digits
        